addons/mrp/models/mrp_unbuild.py

```diff
@@ -121,14 +121,23 @@ def action_unbuild(self):
         for produce_move in produce_moves:
             if produce_move.has_tracking != 'none':
                 original = original_quants.filtered(lambda quant: quant.product_id == produce_move.product_id)
-                if original and not original[0].lot_id:
+                if not original or sum(original.filtered(lambda q: q.lot_id).mapped('qty')) < produce_moves.product_qty:
                     raise UserError(_("You don't have in the stock the required lot/serial number for %s .") % (produce_move.product_id.name,))
-                self.env['stock.move.lots'].create({
-                    'move_id': produce_move.id,
-                    'lot_id': original[0].lot_id.id,
-                    'quantity_done': produce_move.product_uom_qty,
-                    'quantity': produce_move.product_uom_qty
-                })
+                else:
+                    quantity_todo = produce_move.product_qty
+                    for quant in original:
+                        if quantity_todo <= 0:
+                            break
+                        if not quant.lot_id:
```

Comment:
error?

Reply:
we could group quants (not to have twice the same stock.move.lots)

Reply:
Let remove that part quant without lot should not pass there